tests/dbshell/test_mongodb.py

```diff
@@ -0,0 +1,118 @@
+import signal
+from unittest import mock
+
+from django_mongodb.client import DatabaseClient
+
+from django.db import connection
+from django.test import SimpleTestCase
+
+
+class MongoDBDbshellCommandTests(SimpleTestCase):
+    def settings_to_cmd_args_env(self, settings_dict, parameters=None):
+        if parameters is None:
+            parameters = []
+        return DatabaseClient.settings_to_cmd_args_env(settings_dict, parameters)
+
+    def test_fails_with_keyerror_on_incomplete_config(self):
+        with self.assertRaises(KeyError):
+            self.settings_to_cmd_args_env({})
+
+    def test_basic_params_specified_in_settings(self):
+        for options_parameters in [(None, None, None), ("value1", "value2", True)]:
+            with self.subTest(keys=options_parameters):
+                authentication_database, authentication_mechanism, retry_writes = (
+                    options_parameters
+                )
+                if authentication_database is not None:
+                    expected_args = [
+                        "mongosh",
+                        "--host",
+                        "somehost",
+                        "--port",
+                        444,
+                        "--username",
+                        "someuser",
+                        "--password",
+                        "somepassword",
+                        "--retryWrites",
+                        "true",
+                        "somedbname",
+                    ]
+                else:
+                    expected_args = [
+                        "mongosh",
+                        "--host",
+                        "somehost",
+                        "--port",
+                        444,
+                        "--username",
+                        "someuser",
+                        "--password",
+                        "somepassword",
+                        "somedbname",
+                    ]
+
+                self.assertEqual(
+                    self.settings_to_cmd_args_env(
+                        {
+                            "NAME": "somedbname",
+                            "USER": "someuser",
+                            "PASSWORD": "somepassword",
+                            "HOST": "somehost",
+                            "PORT": 444,
+                            "OPTIONS": {
+                                "authenticationDatabase": authentication_database,
+                                "authenticationMechanism": authentication_mechanism,
+                                "retryWrites": retry_writes,
+                            },
+                        }
+                    ),
+                    (expected_args, None),
+                )
+
+    def test_options_override_settings_proper_values(self):
```

Comment:
This test isn't needed. It's for a line in MySQL's client.py: 
```
port = settings_dict["OPTIONS"].get("port", settings_dict["PORT"])
```
MongoDB doesn't have that.